### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
# Django Rest Blog App Backend:

Description: Backend Django Rest Models and Views for API CRUD operations.

Installation Instructions:
1 .Change directory to `webblog/` and make a virtual environment by running `virtualenv env`
2 .Open terminal and start virtual environment by running `source ./env/bin/activate`
3 .run `pip install -r requirements.txt --no-index --find-links file:///tmp/packages` to make requirements
4. run `python manage.py runserver` to start the django server

### CODE REVIEW

File: webblog/accounts/views.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+from django.shortcuts import render
+
+from rest_framework import generics, permissions
+from rest_framework.response import Response
+from knox.models import AuthToken
```
Comment (zubair-arbi) [inline webblog/accounts/views.py]: Use `isort` to order imports

File: webblog/blog/admin.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (zubair-arbi) [inline webblog/blog/admin.py]: remove this line

File: webblog/blog/migrations/0008_blog_owner.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Generated by Django 2.2.5 on 2019-09-13 10:14
```
Comment (zubair-arbi) [inline webblog/blog/migrations/0008_blog_owner.py]: This is your first Pull request. Remove all the migrations and replace them with just one migration

File: webblog/blog/models.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from django.db import models
+
+
+class Blog(models.Model):
+    title = models.CharField(max_length=100, blank=True, default='')
+    description = models.CharField(max_length=1500)
```
Comment (zubair-arbi) [inline webblog/blog/models.py]: We should use `TextField` for `description`.

File: webblog/package.json [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "webblog",
+  "version": "1.0.0",
```
Comment (zubair-arbi) [inline webblog/package.json]: We don't need this file for django project

File: webblog/webblog/settings.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '728cf6a3lzdg3=u1w7odg80&0-wwhc7r&n19c3n5dwm3$uiac('
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (zubair-arbi) [inline webblog/webblog/settings.py]: Split settings to `base.py`, `local.py` (for local development and debugging) and `production.py` (with production settings)

File: webblog/webblog/urls.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+"""
+from django.contrib import admin
+from django.urls import path, include
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+    path('api/', include('blog.urls')),
+    path('', include('frontend.urls')),
+    path('', include('accounts.urls'))
+]
```
Comment (zubair-arbi) [inline webblog/webblog/urls.py]: Make sure we have a newline at the end of each python file

File: webblog/webblog/wsgi.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/howto/deployment/wsgi/
+"""
+
+import os
+
+from django.core.wsgi import get_wsgi_application
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'webblog.settings')
```
Comment (zubair-arbi) [inline webblog/webblog/wsgi.py]: by default it should be pointing towards local settings `webblog.settings.local`

File: webblog/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  module: {
```
Comment (zubair-arbi) [inline webblog/webpack.config.js]: remove this file

**Comment by zubair-arbi** [comment]
Always make sure to add proper information in your pull requests similar to this https://github.com/edly-io/edly-wp-theme/pull/24

File: webblog/blog/admin.py (hunk 2 above)
Comment (Ali-D-Akbar) [reply webblog/blog/admin.py]: done

File: webblog/accounts/views.py (hunk 1 above)
Comment (Ali-D-Akbar) [reply webblog/accounts/views.py]: done

File: webblog/blog/migrations/0008_blog_owner.py (hunk 3 above)
Comment (Ali-D-Akbar) [reply webblog/blog/migrations/0008_blog_owner.py]: done

File: webblog/webblog/urls.py (hunk 7 above)
Comment (Ali-D-Akbar) [reply webblog/webblog/urls.py]: yes sir I've added them now

File: webblog/blog/models.py (hunk 4 above)
Comment (Ali-D-Akbar) [reply webblog/blog/models.py]: done

File: webblog/webblog/wsgi.py (hunk 8 above)
Comment (Ali-D-Akbar) [reply webblog/webblog/wsgi.py]: implemented the local and also set the default too.

File: webblog/webblog/settings.py (hunk 6 above)
Comment (Ali-D-Akbar) [reply webblog/webblog/settings.py]: done

**Comment by asadmanzoor93** [comment]
> Description: Backend Django Rest Models and Views for API CRUD operations.
> 
> Installation Instructions:
> 1 .Change directory to `webblog/` and make a virtual environment by running `virtualenv env`
> 2 .Open terminal and start virtual environment by running `source ./env/bin/activate`
> 3 .run `pip install -r requirements.txt --no-index --find-links file:///tmp/packages` to make requirements
> 4. run `python manage.py runserver` to start the django server

Add Proper Title of PR

**Pull request merged**